lib/game.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+require 'frame'
+
+class Game
+  attr_reader :frame, :scores
+
+  TP = 10
+  TF = 10
+  def initialize
+    @scores = []
+    @frame = 0
+  end
+
+  def roll(pins)
+    raise 'Your game is now complete' if game_over?
+
+    @scores << pins
+  end
+
+  def score
+    result = 0
+    index = 0
+    TF.times do
+      @frame += 1
+      if strike?(index)
+        result += last_frame ? bonus(index) : strike_score(index)
+        index += 1
+      elsif spare?(index)
+        result += last_frame ? bonus(index) : spare_score(index)
+        index += 2
+      else
+        result += frame_score(index)
+        index += 2
+      end
+    end
+    result
+  end
+
+  private
+
+  def game_over?
+    @scores.count >= 21 || @frame >= TF
```

Comment:
nice, stopping the game from finishing here!

Reply:
This method should purely work based on the score array, as the frame variable is not updated until you call the score.
Think about how you would know whether a game has finished or not looking simply at the score array. Is there a way?